src/kyrios.py

```diff
@@ -85,12 +86,15 @@ def installPackage(packageName, package, context):
         return
 
     platformConfig = getPlatformConfig(context,package)
-    packageManagerName = platformConfig['packageManager']
-    packageManager = packageManagers[packageManagerName]
-    logging.debug("found packageManager: '{}': {}".format(packageManagerName, packageManager))
-
-    packageManager.installPackage(packageName, package, context, platformConfig)
-    context['installedPackages'].append(packageName)
+    if 'packageManager' in platformConfig:
+        packageManagerName = platformConfig['packageManager']
+        packageManager = packageManagers[packageManagerName]
+        logging.debug("found packageManager: '{}': {}".format(packageManagerName, packageManager))
+
+        packageManager.installPackage(packageName, package, context, platformConfig)
+        context['installedPackages'].append(packageName)
+    elif not platformConfig.get('installedByDependency', False):
+        logging.warn("Package '{}' is required, but does not specify a `packageManager`, and is not `installedByDependency`.".format(packageName));
```

Comment:
Are you on a version that supports f-strings? If so, use those?

Reply:
If I remember correctly 2.7 is intentionally used here to be compatible with the default version of python found on many operating systems.